File: src/main/java/com/intellij/uiDesigner/core/config/FitNessePluginConfig.java
package com.intellij.uiDesigner.core.config;

import com.intellij.openapi.components.*;
import com.intellij.util.xmlb.XmlSerializerUtil;
import org.jetbrains.annotations.Nullable;

@State(
        name = "FitNessePluginConfig",
        storages = {
                @Storage(file = StoragePathMacros.APP_CONFIG + "/FitNessePluginConfig.xml")
        }
)
public class FitNessePluginConfig implements PersistentStateComponent<FitNessePluginConfig> {

    private String fitnesseRootFolderUrl;
    private String fixtureSuffix;
    private final String DEFAULT_URL = "http://localhost:8025/NextGen";
    private final String DEFAULT_SUFFIX = "Fixture";

    public String getFitnesseRootFolderUrl() {
        return fitnesseRootFolderUrl;
    }

    public String getFixtureSuffix() {
        return fixtureSuffix;
    }

    void setFitnesseRootFolderUrl(String fitnesseRootFolderUrl) {
        this.fitnesseRootFolderUrl = fitnesseRootFolderUrl;
    }

    void setFixtureSuffix(String fixtureSuffix) {
        this.fixtureSuffix = fixtureSuffix;
    }

    @Nullable
    @Override
    public FitNessePluginConfig getState() {
        return this;
    }

    @Override
    public void loadState(FitNessePluginConfig singleFileExecutionConfig) {
        XmlSerializerUtil.copyBean(singleFileExecutionConfig, this);
        setupDefaultValuesIfNeeded();
    }

    private void setupDefaultValuesIfNeeded() {
        if (getFitnesseRootFolderUrl() == null) {
            setFitnesseRootFolderUrl(DEFAULT_URL);
        }
        if (getFixtureSuffix() == null) {
            setFixtureSuffix(DEFAULT_SUFFIX);
        }
    }

    @Nullable
    public static FitNessePluginConfig getInstance() {
        FitNessePluginConfig config = ServiceManager.getService(FitNessePluginConfig.class);
        if (config == null) {
            config = new FitNessePluginConfig();
        }
        config.setupDefaultValuesIfNeeded();
        return config;
    }


}

File: src/main/scala/fitnesse/idea/config/FitNessePluginConfig.scala
package fitnesse.idea.config

import com.intellij.openapi.components.{PersistentStateComponent, ServiceManager, Storage, StoragePathMacros, State}
import com.intellij.util.xmlb.XmlSerializerUtil

object FitNessePluginConfig {
  val DEFAULT_FIT_SERVER_URL = "http://localhost:8025"
  val DEFAULT_FIT_ROOT = "FitnesseRoot"
  val DEFAULT_SUFFIX = "Fixture"

  def getInstance: FitNessePluginConfig = {
    var config = ServiceManager.getService(classOf[FitNessePluginConfig])
    if (config == null) config = new FitNessePluginConfig
    config.setupDefaultValuesIfNeeded()
    config
  }
}

@State(name = "FitNessePluginConfig", storages = Array(new Storage(file = StoragePathMacros.APP_CONFIG + "/FitNessePluginConfig.xml")))
class FitNessePluginConfig(private var fitServerUrl0: String,
                           private var fitnesseRoot0: String,
                           private var fixtureSuffix0: String) extends PersistentStateComponent[FitNessePluginConfig] {
  def this() = this(null, null, null)

  import FitNessePluginConfig._

  def fitServerUrl: String = fitServerUrl0
  def fitServerUrl_=(url: String): Unit = fitServerUrl0 = url

  def fitnesseRoot: String = fitnesseRoot0
  def fitnesseRoot_=(root: String): Unit = fitnesseRoot0 = root

  def fixtureSuffix: String = fixtureSuffix0
  def fixtureSuffix_=(suffix: String): Unit = fixtureSuffix0 = suffix

  override def loadState(config: FitNessePluginConfig): Unit = {
    XmlSerializerUtil.copyBean(config, this)
    setupDefaultValuesIfNeeded()
  }

  private def setupDefaultValuesIfNeeded() = {
    if (fitServerUrl == null) fitServerUrl = DEFAULT_FIT_SERVER_URL
    if (fitnesseRoot == null) fitnesseRoot = DEFAULT_FIT_ROOT
    if (fixtureSuffix == null) fixtureSuffix = DEFAULT_SUFFIX
  }

  override def getState: FitNessePluginConfig = this
}
